src/Microsoft.AspNet.OData.Shared/Routing/ODataRouteConstants.cs

```diff
@@ -67,5 +67,11 @@ public static class ODataRouteConstants
         /// Route data key for the OData optional parameters.
         /// </summary>
         public static readonly string OptionalParameters = typeof(ODataOptionalParameter).FullName;
+
+        /// <summary>
+        /// Route data key that tracks the number of key segments,
+        /// navigation properties and operation parameters in the request URI
+        /// </summary>
+        public static readonly string KeyCount = "{keyCount}";
```

Comment:
>{ [](start = 50, length = 1)

{ } typically used for template, It might be confusing here. If it's only used as the "key" in the dictionary, consider to remove braces and add "OData" as prefix?